src/lib/internal/actions/modal/action.ts

```diff
@@ -0,0 +1,73 @@
+import { isElement, last, noop, sleep } from '$lib/internal/helpers/index.js';
+import type { InteractOutsideEvent } from '../interact-outside/types.js';
+import { useInteractOutside } from '..';
+import type { ModalConfig } from './types.js';
+
+const visibleModals: Element[] = [];
+
+export function useModal(node: HTMLElement, config: ModalConfig) {
+	let unsubInteractOutside = noop;
+
+	function removeNodeFromVisibleModals() {
+		const index = visibleModals.indexOf(node);
+		if (index >= 0) {
+			visibleModals.splice(index, 1);
+		}
+	}
+
+	function update(config: ModalConfig) {
+		const { open, onClose, shouldCloseOnInteractOutside, closeOnInteractOutside } = config;
+
+		sleep(100).then(() => {
```

Comment:
Doesn't tick or sleep(1) suffice here?

Reply:
Nice catch, let me take another look to ensure the tests still pass without this. I know it was added during debugging the nested dialogs, so I may have let it sneak in by accident!

Reply:
Actually, I left it as it was originally in your implementation of handling the nested dialogs:

![CleanShot 2024-01-11 at 10 27 02@2x](https://github.com/melt-ui/melt-ui/assets/64506580/ff2e5b24-4f59-462d-b4aa-2e152051f461)

We can see if the tests still pass with it being 1 or less than 100 though!

Reply:
Okay I removed this and it passed all the tests 🫡